### PR TITLE
Gradle dependencies command handling

### PR DESCRIPTION
**Pull request by atomfrede**
Just for early feedback to know if thats the right direction. Next step would be to add a direct dependency. For the dependency management equivalent I am thinking of using the version catalogs feature, but thats second or third step. 
 

udpates #1862

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/module/infrastructure/secondary/GradleCommandHandler.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import tech.jhipster.lite.module.domain.Indentation;
+import tech.jhipster.lite.module.domain.javabuild.command.AddBuildPluginManagement;
+import tech.jhipster.lite.module.domain.javabuild.command.AddDirectJavaBuildPlugin;
+import tech.jhipster.lite.module.domain.javabuild.command.AddDirectJavaDependency;
+import tech.jhipster.lite.module.domain.javabuild.command.AddJavaDependencyManagement;
+import tech.jhipster.lite.module.domain.javabuild.command.RemoveDirectJavaDependency;
+import tech.jhipster.lite.module.domain.javabuild.command.RemoveJavaDependencyManagement;
+import tech.jhipster.lite.module.domain.javabuild.command.SetVersion;
+
+public class GradleCommandHandler extends JavaDependenciesCommandHandler {
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/infrastructure/secondary/GradleCommandHandler.java:14]: Should be package private

**Review by DamnClin — COMMENTED** [review]
Looks a lot like what I got in mind expect I wasn't expecting that you'll need xml manipulation on one file for gradle

**Comment by atomfrede** [comment]
Yes, xml manipulation should not be needed. Still thinking of a nice way to do it. The more I tempting using rewrite looks.

**Comment by DamnClin** [comment]
> Yes, xml manipulation should not be needed. Still thinking of a nice way to do it. The more I tempting using rewrite looks.

So you should extract an interface instead of an abstract class

**Comment by atomfrede** [comment]
Ah! 🤦 No I got it, because of the match inside the super class. Yes thats totally nonsense.

File: src/main/java/tech/jhipster/lite/module/infrastructure/secondary/JavaDependenciesCommandHandler.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import tech.jhipster.lite.module.domain.Indentation;
+import tech.jhipster.lite.module.domain.javabuild.command.AddBuildPluginManagement;
+import tech.jhipster.lite.module.domain.javabuild.command.AddDirectJavaBuildPlugin;
+import tech.jhipster.lite.module.domain.javabuild.command.AddDirectJavaDependency;
+import tech.jhipster.lite.module.domain.javabuild.command.AddJavaDependencyManagement;
+import tech.jhipster.lite.module.domain.javabuild.command.RemoveDirectJavaDependency;
+import tech.jhipster.lite.module.domain.javabuild.command.RemoveJavaDependencyManagement;
+import tech.jhipster.lite.module.domain.javabuild.command.SetVersion;
+
+public interface JavaDependenciesCommandHandler {
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/infrastructure/secondary/JavaDependenciesCommandHandler.java:22]: The can be package private

**Comment by murdos** [comment]
@atomfrede : I'm closing this draft since a few PR for gradle support have already been merged, and https://github.com/jhipster/jhipster-lite/issues/1862 has been fixed

**Pull request closed without merging**